### ISSUE TITLE
Text cannot be parsed to a Duration

### ISSUE THREAD

**Issue by jamieb22** [ical4j/ical4j#366]
**Describe the bug**

Below is iCal event received from MS Exchange. iCal4j wont parse it despite leniency settings enabled. The server is MS Office 365. Error at line 42:Text cannot be parsed to a Duration

```
BEGIN:VCALENDAR
METHOD:PUBLISH
PRODID:Microsoft Exchange Server 2010
VERSION:2.0
BEGIN:VTIMEZONE
TZID: 1
BEGIN:STANDARD
DTSTART:16010101T000000
TZOFFSETFROM:+0200
TZOFFSETTO:+0200
END:STANDARD
BEGIN:DAYLIGHT
DTSTART:16010101T000000
TZOFFSETFROM:+0200
TZOFFSETTO:+0200
END:DAYLIGHT
END:VTIMEZONE
BEGIN:VEVENT
ORGANIZER;CN=Joe Blogs:MAILTO:joe@company.co.za
RRULE:FREQ=YEARLY;UNTIL=20361231T060000Z;INTERVAL=1;BYMONTHDAY=12;BYMONTH=1
UID:1491ebe4-74bc-45da-95f1-8c6858658d32
SUMMARY;LANGUAGE=en-ZA:Joe Birthday
DTSTART;TZID= 1:20130112T080000
DTEND;TZID= 1:20130112T090000
CLASS:PUBLIC
PRIORITY:5
DTSTAMP:20180116T054539Z
TRANSP:TRANSPARENT
STATUS:CONFIRMED
SEQUENCE:0
X-MICROSOFT-CDO-APPT-SEQUENCE:0
X-MICROSOFT-CDO-OWNERAPPTID:2110872603
X-MICROSOFT-CDO-BUSYSTATUS:FREE
X-MICROSOFT-CDO-INTENDEDSTATUS:BUSY
X-MICROSOFT-CDO-ALLDAYEVENT:FALSE
X-MICROSOFT-CDO-IMPORTANCE:1
X-MICROSOFT-CDO-INSTTYPE:1
X-MICROSOFT-DONOTFORWARDMEETING:FALSE
X-MICROSOFT-DISALLOW-COUNTER:FALSE
BEGIN:VALARM
DESCRIPTION:REMINDER
TRIGGER;RELATED=START:P
ACTION:DISPLAY
END:VALARM
END:VEVENT
END:VCALENDAR
```

**To Reproduce**

Try to parse above with:

```
CompatibilityHints.setHintEnabled(CompatibilityHints.KEY_NOTES_COMPATIBILITY,true);
CompatibilityHints.setHintEnabled(CompatibilityHints.KEY_OUTLOOK_COMPATIBILITY,true);
CompatibilityHints.setHintEnabled(CompatibilityHints.KEY_RELAXED_PARSING,true);
CompatibilityHints.setHintEnabled(CompatibilityHints.KEY_RELAXED_UNFOLDING,true);
CompatibilityHints.setHintEnabled(CompatibilityHints.KEY_RELAXED_VALIDATION,true);
CompatibilityHints.setHintEnabled(CompatibilityHints.KEY_VCARD_COMPATIBILITY,true);
```
```

Caused by: java.time.format.DateTimeParseException: Text cannot be parsed to a Duration
	at java.time.Duration.parse(Duration.java:412)
	at net.fortuna.ical4j.model.TemporalAmountAdapter.parse(TemporalAmountAdapter.java:121)
	at net.fortuna.ical4j.model.property.Trigger.setValue(Trigger.java:270)
	at net.fortuna.ical4j.model.property.Trigger.<init>(Trigger.java:165)
	at net.fortuna.ical4j.model.property.Trigger$Factory.createProperty(Trigger.java:315)
	at net.fortuna.ical4j.model.PropertyBuilder.build(PropertyBuilder.java:48)
	at net.fortuna.ical4j.data.DefaultContentHandler.endProperty(DefaultContentHandler.java:132)
	at net.fortuna.ical4j.data.CalendarParserImpl$PropertyParser.parse(CalendarParserImpl.java:292)
	at net.fortuna.ical4j.data.CalendarParserImpl$PropertyParser.access$1100(CalendarParserImpl.java:224)
	at net.fortuna.ical4j.data.CalendarParserImpl$PropertyListParser.parse(CalendarParserImpl.java:211)
	at net.fortuna.ical4j.data.CalendarParserImpl$ComponentParser.parse(CalendarParserImpl.java:427)
	at net.fortuna.ical4j.data.CalendarParserImpl$ComponentParser.access$900(CalendarParserImpl.java:404)
	at net.fortuna.ical4j.data.CalendarParserImpl$PropertyListParser.parse(CalendarParserImpl.java:209)
	at net.fortuna.ical4j.data.CalendarParserImpl$ComponentParser.parse(CalendarParserImpl.java:427)
	at net.fortuna.ical4j.data.CalendarParserImpl$ComponentParser.access$900(CalendarParserImpl.java:404)
	at net.fortuna.ical4j.data.CalendarParserImpl$PropertyListParser.parse(CalendarParserImpl.java:209)
	at net.fortuna.ical4j.data.CalendarParserImpl.parseCalendar(CalendarParserImpl.java:115)
	at net.fortuna.ical4j.data.CalendarParserImpl.parseCalendarList(CalendarParserImpl.java:180)
	at net.fortuna.ical4j.data.CalendarParserImpl.parse(CalendarParserImpl.java:149)
	... 22 common frames omitted

```
**Expected behavior**

If relaxed parsing is enabled, iCal4j should parse the file. 

**Environment (please complete the following information):**
Linux, Java 8

**Additional context**
Data received from O365.

**Comment by jamieb22** [comment]
To ignore the bad property and continue.. I modified CalendarParserImpl to catch Exceptions if KEY_RELAXED_PARSING is enabled.

```
public void parse(final StreamTokenizer tokeniser, Reader in,
                          final ContentHandler handler) throws IOException, ParseException,
                URISyntaxException, ParserException {

            assertToken(tokeniser, in, StreamTokenizer.TT_WORD);

            while (/*
             * !Component.BEGIN.equals(tokeniser.sval) &&
             */!Component.END.equals(tokeniser.sval)) {
                // check for timezones observances or vevent/vtodo alarms..
                if (Component.BEGIN.equals(tokeniser.sval)) {
                    // modification: caught parsing error to ensure ical4j will do best effort to process file
                    try {
                        componentParser.parse(tokeniser, in, handler);
                    }  catch (Exception e) {
                            if (!CompatibilityHints.isHintEnabled(CompatibilityHints.KEY_RELAXED_PARSING))
                                throw new ParserException("Invalid property name", getLineNumber(tokeniser, in));
                    }
                } else if (tokeniser.sval != null) {
                    // Jamie modification: caught parsing error to ensure ical4j will do best effort to process file
                    try {
                        propertyParser.parse(tokeniser, in, handler);
                    } catch (Exception e) {
                        if (!CompatibilityHints.isHintEnabled(CompatibilityHints.KEY_RELAXED_PARSING))
                            throw new ParserException("Invalid property name", getLineNumber(tokeniser, in)); 
                    }
                } else if (!CompatibilityHints.isHintEnabled(CompatibilityHints.KEY_RELAXED_PARSING)) {
                    throw new ParserException("Invalid property name", getLineNumber(tokeniser, in));
                }
                absorbWhitespace(tokeniser, in, false);
                // assertToken(tokeniser, StreamTokenizer.TT_WORD);
            }
        }
    }
```

**Comment by benfortuna** [comment]
@jamieb22 do you know if the intention was to create a reminder of zero duration (i.e. 'P' => 'PT0S'). If so we can probably support this in the `Trigger` class.

**Comment by jamieb22** [comment]
We parse all events in MS Exchange. We don't really know how they were constructed.

**Comment by benfortuna** [comment]
Fix applied in: #368
